dcr: Panic when clicking staking with a new wallet.

Newly created cryptopower on testnet on master. Restore a decred wallet then click on staking before doing anything else.

<details closed>
<summary>panic</summary>

```
2023-12-14 14:22:18.181 [INF] WLLT: Upgrading database from version 1 to 25
2023-12-14 14:22:20.039 [INF] DCR: Opened wallet
2023-12-14 14:22:20.040 [INF] SHWL: Created Wallet
panic: runtime error: slice bounds out of range [:-1]

goroutine 56 [running]:
github.com/crypto-power/cryptopower/ui/page/components.formatBalance({{{0x0, 0x0}, {0x6cd, 0xf41b7}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/page/components/coinformat.go:46 +0x6da
github.com/crypto-power/cryptopower/ui/page/components.LayoutBalanceWithUnitSizeBoldText(...)
        /home/joe/git/cryptopower/ui/page/components/coinformat.go:130
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).stakePriceSection.func1.1.1.1.1.1.2.1({{{0x0, 0x0}, {0x6cd, 0xf41b7}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_info.go:86 +0x150
gioui.org/layout.Direction.Layout(0x8, {{{0x0, 0x0}, {0x6cd, 0xf41b7}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:196 +0x155
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).stakePriceSection.func1.1.1.1.1.1.2({{{0x0, 0x0}, {0x6cd, 0xf41b7}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_info.go:80 +0xab
gioui.org/layout.Flex.Layout({0x90?, 0x8c?, 0x9f?, 0xc0?}, {{{0x0, 0x0}, {0x722, 0xf41b7}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/flex.go:99 +0x2d5
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).stakePriceSection.func1.1.1.1.1.1({{{0x0, 0x0}, {0x722, 0xf41b7}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_info.go:73 +0x254
gioui.org/layout.Inset.Layout({0x27?, 0x0?, 0x5dfba70?, 0xc0?}, {{{0x0, 0x0}, {0x722, 0xf41b7}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).stakePriceSection.func1.1.1.1.1({{{0x0, 0x0}, {0x722, 0xf41c3}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_info.go:72 +0xb5
gioui.org/layout.Flex.Layout({0x1a?, 0x0?, 0x0?, 0x0?}, {{{0x0, 0x0}, {0x722, 0xf41c3}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/flex.go:99 +0x2d5
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).stakePriceSection.func1.1.1.1({{{0x0, 0x0}, {0x722, 0xf41c3}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_info.go:67 +0x215
gioui.org/layout.Flex.Layout({0xa?, 0x0?, 0x0?, 0x0?}, {{{0x722, 0x0}, {0x722, 0xf41c3}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/flex.go:99 +0x2d5
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).stakePriceSection.func1.1.1({{{0x722, 0x0}, {0x722, 0xf41c3}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_info.go:113 +0x237
gioui.org/layout.Inset.Layout({0x5dfba70?, 0xc0?, 0x25?, 0x0?}, {{{0x722, 0x0}, {0x722, 0xf41c3}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).stakePriceSection.func1.1({{{0x722, 0x0}, {0x722, 0xf41ce}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_info.go:64 +0x89
gioui.org/layout.Flex.Layout({0x9?, 0x0?, 0x0?, 0x0?}, {{{0x722, 0x0}, {0x722, 0xf4200}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/flex.go:99 +0x2d5
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).stakePriceSection.func1({{{0x722, 0x0}, {0x722, 0xf4200}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_info.go:59 +0x23b
gioui.org/layout.Inset.Layout({0x5dfba70?, 0xc0?, 0x10?, 0x0?}, {{{0x722, 0x0}, {0x722, 0xf4200}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).stakePriceSection.(*Page).pageSections.func2.1({{{0x752, 0x0}, {0x752, 0xf4230}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_overview.go:241 +0x89
gioui.org/layout.Stack.Layout({0x8?}, {{{0x752, 0x0}, {0x752, 0xf4230}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/stack.go:60 +0x175
github.com/crypto-power/cryptopower/ui/cryptomaterial.Card.Layout.func1({{{0x752, 0x0}, {0x752, 0xf4230}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/card.go:64 +0x18e
gioui.org/layout.Inset.Layout({0x18250a0?, 0x0?, 0xf?, 0x0?}, {{{0x752, 0x0}, {0x752, 0xf4230}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/cryptomaterial.Card.Layout({{0x0, 0x0, 0x0, 0x0}, {0xff, 0xff, 0xff, 0xff}, {0xf3, 0xf5, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/cryptomaterial/card.go:63 +0x109
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).stakePriceSection.(*Page).pageSections.func2({{{0x752, 0x0}, {0x752, 0xf4230}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_overview.go:239 +0xfe
gioui.org/layout.Inset.Layout({0x0?, 0x0?, 0x0?, 0x0?}, {{{0x752, 0x0}, {0x752, 0xf4230}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).pageSections(...)
        /home/joe/git/cryptopower/ui/page/staking/stake_overview.go:238
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).stakePriceSection(0xc003b36780, {{{0x752, 0x0}, {0x752, 0xf4240}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_info.go:58 +0x109
gioui.org/layout.Flex.Layout({0x34?, 0x0?, 0x0?, 0x0?}, {{{0x752, 0x0}, {0x752, 0xf4240}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/flex.go:99 +0x2d5
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).layoutDesktop.func1({{{0x752, 0x0}, {0x752, 0xf4240}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/page/staking/stake_overview.go:203 +0x23b
gioui.org/layout.(*List).Layout(0xc000326e50, {{{0x752, 0x0}, {0x752, 0xf4240}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/list.go:122 +0x178
gioui.org/widget/material.ListStyle.Layout({0xc000326e00, {0xc000326e00, {0x40000000, 0x40000000, {0x0, 0x0, 0x0, 0x0}}, {0x42180000, 0x40c00000, ...}}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/widget/material/list.go:290 +0x1d6
github.com/crypto-power/cryptopower/ui/cryptomaterial.ListStyle.Layout(...)
        /home/joe/git/cryptopower/ui/cryptomaterial/list_scroll.go:39
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).layoutDesktop(0xc003b36780, {{{0x0, 0x0}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_overview.go:202 +0x1b0
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).Layout.func3({{{0x0, 0x0}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_overview.go:194 +0x78
gioui.org/layout.Stack.Layout({0x70?}, {{{0x75c, 0x399}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/stack.go:78 +0x478
github.com/crypto-power/cryptopower/ui/page/staking.(*Page).Layout(0xc003b36780, {{{0x75c, 0x399}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x0, 0x0}, {0xc156c2262e723475, ...}, ...})
        /home/joe/git/cryptopower/ui/page/staking/stake_overview.go:197 +0x32e
github.com/crypto-power/cryptopower/ui/page/components.DisablePageWithOverlay.func1({{{0x75c, 0x399}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/components/components.go:885 +0x7d
gioui.org/layout.Stack.Layout({0x9a?}, {{{0x0, 0x0}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/stack.go:78 +0x478
github.com/crypto-power/cryptopower/ui/page/components.DisablePageWithOverlay(0xc00354f4a0, {0x211f218?, 0xc003b36780}, {{{0x0, 0x0}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x2110700, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/page/components/components.go:879 +0x24d
github.com/crypto-power/cryptopower/ui/page/root.(*SingleWalletMasterPage).layoutDesktop.func1.1.1.1({{{0x0, 0x0}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/root/main_page.go:426 +0x2d1
github.com/crypto-power/cryptopower/ui/cryptomaterial.(*SlideAction).TransformLayout(0xc005e9a000, {{{0x0, 0x0}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/cryptomaterial/slide_action.go:161 +0x495
github.com/crypto-power/cryptopower/ui/cryptomaterial.(*SegmentedControl).Layout.func1.2.1.1({{{0x0, 0x0}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/segmented_control.go:103 +0x45
github.com/crypto-power/cryptopower/ui/cryptomaterial.(*SlideAction).DragLayout(0xc005e9a000, {{{0x0, 0x0}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/cryptomaterial/slide_action.go:100 +0x155
github.com/crypto-power/cryptopower/ui/cryptomaterial.(*SegmentedControl).Layout.func1.2.1({{{0x0, 0x0}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/segmented_control.go:102 +0x8b
gioui.org/layout.Inset.Layout({0x4556?, 0x0?, 0xfa?, 0x0?}, {{{0x0, 0x0}, {0x75c, 0x399}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/cryptomaterial.(*SegmentedControl).Layout.func1.2({{{0x0, 0x0}, {0x75c, 0x3a9}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/segmented_control.go:98 +0xb2
gioui.org/layout.Flex.Layout({0x0?, 0x0?, 0x0?, 0x0?}, {{{0x0, 0x0}, {0x75c, 0x3cb}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/flex.go:99 +0x2d5
github.com/crypto-power/cryptopower/ui/cryptomaterial.(*SegmentedControl).Layout.func1({{{0x0, 0x0}, {0x75c, 0x3cb}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/segmented_control.go:90 +0x1d7
gioui.org/layout.Inset.Layout({0x3c?, 0x0?, 0x4?, 0x0?}, {{{0x0, 0x0}, {0x75c, 0x3cb}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/cryptomaterial.UniformPadding({{{0x0, 0x0}, {0x77c, 0x3e3}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/cryptomaterial/util.go:163 +0x1d9
github.com/crypto-power/cryptopower/ui/cryptomaterial.(*SegmentedControl).Layout(0x0?, {{{0x0, 0x0}, {0x77c, 0x3e3}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/cryptomaterial/segmented_control.go:86 +0xb0
github.com/crypto-power/cryptopower/ui/page/root.(*SingleWalletMasterPage).layoutDesktop.func1.1.1({{{0x0, 0x0}, {0x77c, 0x3e3}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/root/main_page.go:416 +0xce
gioui.org/layout.Inset.Layout({0x5c394d0?, 0xc0?, 0x48?, 0x0?}, {{{0x0, 0x0}, {0x77c, 0x3e3}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/page/root.(*SingleWalletMasterPage).layoutDesktop.func1.1({{{0x0, 0x0}, {0x77c, 0x3f3}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/root/main_page.go:415 +0x89
gioui.org/layout.Flex.Layout({0xb?, 0x0?, 0x0?, 0x0?}, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/flex.go:99 +0x2d5
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1.1.2.1.1.1({{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:88 +0x65
gioui.org/layout.Direction.Layout(0x0, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:196 +0x155
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1.1.2.1.1({{{0x77c, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:87 +0xc6
gioui.org/layout.Inset.Layout({0x39ff5a0?, 0xc0?, 0x69ee67?, 0x0?}, {{{0x77c, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1.1.2.1({{{0x77c, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:85 +0xe9
github.com/crypto-power/cryptopower/ui/cryptomaterial.Border.Layout({{0x0, 0x0, 0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, 0x0}, {{{0x77c, ...}, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/cryptomaterial/border.go:21 +0x98
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1.1.2({{{0x77c, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:83 +0x1b6
gioui.org/layout.Stack.Layout({0xe3?}, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/stack.go:60 +0x175
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1.1({{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:46 +0x2aa
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1({{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:105 +0x219
gioui.org/layout.Inset.Layout({0x7c?, 0x0?, 0x100?, 0x0?}, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1({{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:44 +0xe9
gioui.org/layout.Direction.Layout(0x0, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:196 +0x155
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout({0xfffffffffffffffe, 0xfffffffffffffffe, 0x1, {0x0, 0x0, 0x0, 0x0}, 0x0, {{0x0, 0x0, ...}, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:42 +0xe5
github.com/crypto-power/cryptopower/ui/page/root.(*SingleWalletMasterPage).layoutDesktop.func1({{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/root/main_page.go:410 +0x285
gioui.org/layout.Stack.Layout({0x0?}, {{{0x0, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/stack.go:78 +0x478
github.com/crypto-power/cryptopower/ui/page/root.(*SingleWalletMasterPage).layoutDesktop(0xc00767c100, {{{0x0, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...})
        /home/joe/git/cryptopower/ui/page/root/main_page.go:399 +0x12b
github.com/crypto-power/cryptopower/ui/page/root.(*SingleWalletMasterPage).Layout(0x0?, {{{0x0, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...})
        /home/joe/git/cryptopower/ui/page/root/main_page.go:395 +0x38
gioui.org/layout.Flex.Layout({0x5?, 0x0?, 0x0?, 0x0?}, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/flex.go:134 +0x735
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1.1.2.1.1.1({{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:88 +0x65
gioui.org/layout.Direction.Layout(0x0, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:196 +0x155
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1.1.2.1.1({{{0x77c, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:87 +0xc6
gioui.org/layout.Inset.Layout({0x3a01b68?, 0xc0?, 0x69ee67?, 0x0?}, {{{0x77c, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1.1.2.1({{{0x77c, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:85 +0xe9
github.com/crypto-power/cryptopower/ui/cryptomaterial.Border.Layout({{0x0, 0x0, 0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, 0x0}, {{{0x77c, ...}, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/cryptomaterial/border.go:21 +0x98
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1.1.2({{{0x77c, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:83 +0x1b6
gioui.org/layout.Stack.Layout({0x8a?}, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/stack.go:60 +0x175
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1.1({{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:46 +0x2aa
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1.1({{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:105 +0x219
gioui.org/layout.Inset.Layout({0xb81800?, 0xc0?, 0xf0480?, 0x10000c0?}, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:169 +0x39a
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout.func1({{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:44 +0xe9
gioui.org/layout.Direction.Layout(0x0, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/layout.go:196 +0x155
github.com/crypto-power/cryptopower/ui/cryptomaterial.LinearLayout.Layout({0xfffffffffffffffe, 0xfffffffffffffffe, 0x1, {0x0, 0x0, 0x0, 0x0}, 0x0, {{0x0, 0x0, ...}, ...}, ...}, ...)
        /home/joe/git/cryptopower/ui/cryptomaterial/linearlayout.go:42 +0xe5
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).layoutDesktop.func1({{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/page/root/home_page.go:361 +0x265
gioui.org/layout.Stack.Layout({0xb0?}, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/stack.go:78 +0x478
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).layoutDesktop(0xc00709c340, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...})
        /home/joe/git/cryptopower/ui/page/root/home_page.go:355 +0x12b
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).Layout(0x0?, {{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...})
        /home/joe/git/cryptopower/ui/page/root/home_page.go:351 +0x78
github.com/crypto-power/cryptopower/ui.(*Window).prepareToDisplayUI.func2({{{0x0, 0x0}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, 0x207a2d8ea8, ...}, ...})
        /home/joe/git/cryptopower/ui/window.go:284 +0x11f
gioui.org/layout.Stack.Layout({0xd0?}, {{{0x77c, 0x41e}, {0x77c, 0x41e}}, {0x3f800000, 0x3f800000}, {0x2110700, 0xc00197d2c8}, {0xc156c2262e723475, ...}, ...}, ...)
        /home/joe/go/pkg/mod/gioui.org@v0.1.0/layout/stack.go:60 +0x175
github.com/crypto-power/cryptopower/ui.(*Window).prepareToDisplayUI(0xc0030e30e0, {{0xc156c2262e723475, 0x207a2d8ea8, 0x2e909c0}, {0x3f800000, 0x3f800000}, {0x77c, 0x41e}, {0x0, 0x0, ...}, ...})
        /home/joe/git/cryptopower/ui/window.go:299 +0x3ed
github.com/crypto-power/cryptopower/ui.(*Window).handleFrameEvent(0xc0030e30e0, {{0xc156c2262e723475, 0x207a2d8ea8, 0x2e909c0}, {0x3f800000, 0x3f800000}, {0x77c, 0x41e}, {0x0, 0x0, ...}, ...})
        /home/joe/git/cryptopower/ui/window.go:237 +0x273
github.com/crypto-power/cryptopower/ui.(*Window).HandleEvents(0xc0030e30e0)
        /home/joe/git/cryptopower/ui/window.go:200 +0x2b6
main.main.func3()
        /home/joe/git/cryptopower/main.go:102 +0x17
created by main.main in goroutine 1
        /home/joe/git/cryptopower/main.go:100 +0x4c6

```
</details>

This panic may be related, but after sync is complete, if I click on "Transactions" and then "Staking Activities" there is also a panic.

```
2023-12-14 14:48:42.305 [INF] SHWL: Exchange sync: update complete for Flypme
2023-12-14 14:48:42.305 [INF] SHWL: Exchange sync: completed
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x14623ab]

goroutine 29 [running]:
github.com/crypto-power/cryptopower/ui/cryptomaterial.(*DropDown).SelectedIndex(...)
        /home/joe/git/cryptopower/ui/cryptomaterial/dropdown.go:151
github.com/crypto-power/cryptopower/ui/page/transaction.(*TransactionsPage).HandleUserInteractions(0xc003b0ea00)
        /home/joe/git/cryptopower/ui/page/transaction/transactions_page.go:525 +0x3eb
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).HandleUserInteractions(0xc00355e9c0)
        /home/joe/git/cryptopower/ui/page/root/home_page.go:186 +0x56
github.com/crypto-power/cryptopower/ui.(*Window).handleFrameEvent(0xc0030692c0, {{0xc156c3b09b6cd540, 0x33b64a97d, 0x2e8a960}, {0x3f800000, 0x3f800000}, {0x77c, 0x41e}, {0x0, 0x0, ...}, ...})
        /home/joe/git/cryptopower/ui/window.go:229 +0x194
github.com/crypto-power/cryptopower/ui.(*Window).HandleEvents(0xc0030692c0)
        /home/joe/git/cryptopower/ui/window.go:200 +0x2b6
main.main.func3()
        /home/joe/git/cryptopower/main.go:102 +0x17
created by main.main in goroutine 1
        /home/joe/git/cryptopower/main.go:100 +0x4c6
```

--- a/ui/page/components/coinformat.go
+++ b/ui/page/components/coinformat.go
@@ -43,6 +43,10 @@ func formatBalance(gtx C, l *load.Load, amount string, mainTextSize unit.Sp, col
 		stopIndex = len(amount)
 	}
 
+	if startIndex > stopIndex || stopIndex <= 0 || stopIndex > len(amount) {
+		return D{}
+	}
+
 	mainText, subText, unitText := amount[:startIndex], amount[startIndex:stopIndex], amount[stopIndex:]
 
 	subTextSize := unit.Sp(float32(mainTextSize) * defaultScale)

--- a/ui/page/components/coinformat_test.go
+++ b/ui/page/components/coinformat_test.go
@@ -1,0 +1,40 @@
+package components
+
+import (
+	"image/color"
+	"testing"
+
+	"gioui.org/op"
+	"github.com/crypto-power/cryptopower/ui/assets"
+	"github.com/crypto-power/cryptopower/ui/cryptomaterial"
+	"github.com/crypto-power/cryptopower/ui/load"
+)
+
+// TestFormatBalance currently only tests for panics.
+func TestFormatBalance(t *testing.T) {
+	th := cryptomaterial.NewTheme(assets.FontCollection(), assets.DecredIcons, false)
+	ld := &load.Load{Theme: th}
+	gtx := C{Ops: new(op.Ops)}
+	tests := []struct {
+		name, amount string
+	}{{
+		name:   "normal",
+		amount: "1 DCR",
+	}, {
+		name:   "one d",
+		amount: "1 D",
+	}, {
+		name:   "decimals",
+		amount: "1.1 DCR",
+	}, {
+		name: "blank amount",
+	}, {
+		name:   "just the coin",
+		amount: "DCR",
+	}}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			formatBalance(gtx, ld, test.amount, 1, color.NRGBA{}, false, false)
+		})
+	}
+}

--- a/ui/page/staking/stake_info.go
+++ b/ui/page/staking/stake_info.go
@@ -78,7 +78,7 @@ func (pg *Page) stakePriceSection(gtx C) D {
 										}),
 										layout.Rigid(func(gtx C) D {
 											return layout.Center.Layout(gtx, func(gtx C) D {
-												if pg.dcrWallet.IsSyncing() || pg.dcrWallet.IsRescanning() || !pg.isTicketsPurchaseAllowed() {
+												if !pg.dcrWallet.IsSynced() || pg.dcrWallet.IsRescanning() || !pg.isTicketsPurchaseAllowed() {
 													title := pg.Theme.Label(values.TextSize16, values.String(values.StrLoadingPrice))
 													title.Color = col
 													return title.Layout(gtx)

